osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -63,8 +63,10 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
             double drainRate = beatmap.Difficulty.DrainRate;
 
             int maxCombo = beatmap.HitObjects.Count;
-            // Add the ticks + tail of the slider. 1 is subtracted because the head circle would be counted twice (once for the slider itself in the line above)
-            maxCombo += beatmap.HitObjects.OfType<Slider>().Sum(s => s.NestedHitObjects.Count - 1);
+            // Add the ticks + tail of the slider
+            // 1 is subtracted because the head circle would be counted twice (once for the slider itself in the line above)
+            // an additional 1 is subtracted if only nested objects are judged because the hit result of the entire slider would not contribute to combo
+            maxCombo += beatmap.HitObjects.OfType<Slider>().Sum(s => s.NestedHitObjects.Count - 1 - (s.OnlyJudgeNestedObjects ? 1 : 0));
```

Comment:
Better than this is to go over all hitobjects and their nested objects and look for judgements where `judgement => judgement.MaxResult.AffectsCombo()`